generate_release_notes.py

```diff
@@ -125,180 +149,258 @@ def get_repo_list(github_org, github):
                     try:
                         pr_list = found_commit.get_pulls()
                         if pr_list:
                             prr_list.update(list(pr_list))
-                    except Exception as e:
-                        logging.warning(f"Failed to retrieve pull requests for commit {found_commit.sha}: {e}")
+                    except GithubException as error:
+                        logging.warning(
+                            "Failed to retrieve pull requests for commit %s: %s",
+                            found_commit.sha, error)
                         skipped_prs.append(f"Commit {found_commit.sha} in repo {github_repo} could not retrieve PRs")
                     else:
-                        commit_url = found_commit.commit.html_url[:-34]
-                        commit_title = "{} (Commit [{}])".format(found_commit.commit.message.strip().split('\n', 1)[0], found_commit.commit.sha[:6])
+                        commit_title = "{} (Commit [{}])".format(
+                            found_commit.commit.message.strip().split('\n', 1)[0], found_commit.commit.sha[:6]
+                        )
                         commit_only.add(commit_title)
-                issue_titles_bugs, issue_titles_epics, issue_titles_other = [], [], []
+
+                issue_titles_epics, issue_titles_other, github_issues = set(), [], []
                 pull_requests_missing_epics = set()
 
-                # Check Jira keys and GitHub issues in PRs and fetch Epic details
                 if prr_list:
                     for pr in prr_list:
                         try:
-                            jira_keys, github_issues = get_issues_from_pr(repo, pr.number)
+                            jira_keys, pr_github_issues = get_issues_from_pr(repo, pr.number)
                             if jira_keys:
                                 for jira_key in jira_keys:
-                                    jira_issue = get_jira_issue(jira_key, args.jira_url, args.jira_email, args.jira_token)
+                                    jira_issue = get_jira_issue(jira_key, parsed_args.jira_url, parsed_args.jira_email, parsed_args.jira_token)
                                     if jira_issue:
-                                        epic_key, epic_title, epic_description = get_parent_epic(jira_issue, args.jira_url, args.jira_email, args.jira_token)
+                                        epic_key, epic_title, epic_description = get_parent_epic(
+                                            jira_issue, parsed_args.jira_url, parsed_args.jira_email, parsed_args.jira_token)
                                         if epic_title:
-                                            issue_titles_epics.append(f"{epic_key} - {epic_title}: {epic_description}")
+                                            issue_titles_epics.add(f"{epic_key} - {epic_title}: {epic_description}")
+                                            pr_mapping[epic_key] = pr.number
```

Comment:
This does not account for multiple PRs making up  a single epic. As currently written each epic will only be associated with a single PR number and it will be the last PR for that epic. This code needs to be updates to have the `pr_mapping` be a dictionary of `epic_keys` to array of pr numbers

Reply:
Updated